src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -667,6 +668,7 @@ def add_electrical_series(
     rate = calculate_regular_series_rate(series=timestamps)  # Returns None if it is not regular
 
     if rate:
+        starting_time = starting_time + timestamps[0]
```

Comment:
It remains to be discussed what to do with the `starting_time` parameter in the case of multiple segment extractors. Right now, this parameter is working as a global shift for the timestamps across all segments.

Reply:
Sooo just to clarify, how are the timestamps from the later segments handled by spikeinterface?

But yeah the intention of the global `starting_time` is to shift all acquisition times by that offset typical of cross-device synchronization.

Reply:
Every segment has either timestamps or t_start assigned to them. If they don't, they produce time by using the sampling frequency and the number of frames.

Reply:
>  If they don't, they produce time by using the sampling frequency and the number of frames.

And applying the shift of subsequent segments under the assumption that all previous ones were contiguous?

Reply:
No. No time shifting is done if neither of those things is provided. So segments with equal frames would produce identical times with `get_times()` in the absence of both times and t_start.

Reply:
But isn't the notion of a 'segment' in neo to represent non-simultaneous recording intervals?

If multiple segments like this share start times, it would result in incorrect data being written by default to the NWBFile (granted, so would assuming perfectly contiguous data), which I would like to avoid.

In practice, do format-specific multi-segment IO's like Neuralynx track and set these start times?

Reply:
What do you mean they share?
Just to clairfy, the 't_start' is different and is not the one that is set with the argument here. 

Concerning your question at the end, what I have seen is that they do.

Reply:
> What do you mean they share?

From what you said before

> So segments with equal frames would produce identical times with get_times()

so they 'share' the the same times even though they represent disjoint non-simultaneous recordings.

> Concerning your question at the end, what I have seen is that they do.

What we should do then is probably at least check and throw a warning if the segments have unset/default/identical start times. But in practice, a supported format ought to set it appropriately every time so we can write the correct starting times to NWB.

Reply:
Ah, yes, the confusion arose because in spikeinterface that case corresponds to recordings that don't have times so there is nothing to share. But yes, I agree it does not make sense for two segments of a multi segment extractor to produce the same result with `get_times()`.

Reply:
Cool, and we can test for that here right?

Reply:
Do you think it should just issue a warning or outright throw an error?

Reply:
I rather tests and issue warnings for this on `spikeinterface`. I will discuss this with Samuel once he is back.